content/chainguard/migration-guides/tutorial/index.md

```diff
@@ -0,0 +1,732 @@
+* Chainguard Images are based on `glibc` and our packages cannot be mixed with Alpine packages
+* The entrypoint on Chainguard Images is likely to be different to other common images (due to the
+  lack of a shell) which can be confusing -- for example shell commands may get unexpectedly run by
+  the Python interpreter
+
+---
+
+The application in question is [identidock](https://github.com/using-docker/identidock). This
+application was written for the book [Using
+Docker](https://learning.oreilly.com/library/view/using-docker/9781491915752/) about ten years ago,
+which shows that we can still migrate software of this age to a new container while realizing the
+benefits of a no-to-low CVE count. The application itself will create
+[identicons](https://en.wikipedia.org/wiki/Identicon) for a user name, similar to what [GitHub
+generates for users with no avatar](https://github.blog/2013-08-14-identicons/). It was designed at
+the time to demonstrate a "microservices" approach, and as such it's made up of 3 services:
+
+
+* The main identidock service, which takes the requests and talks to the
+  [dnmonster](https://github.com/amouat/dnmonster) service and the redis cache
+* A NodeJS application which creates the identicons
+* Redis which is used as a simple cache
+
+The services are put together as shown in the below diagram. The user only talks to the identidock service. The
+identidock service will first check the cache to see if it has already created an identicon for the
+input and if not requests a new identicon from the dnmonster service. The identicon is then returned
+to the user and saved to the cache if required.
+
+![Diagram of Identidock Architecture](arch.png)
+
+The book walked through using various orchestrators to deploy the application, some of which have
+since fallen out of usage (anyone remember [fleet](https://github.com/coreos/fleet)?). For the sake
+of this tutorial, we'll use Docker Compose, which is arguably the simplest surviving orchestrator
+covered in the book.
+
+The first task was to get the 10-year-old application building and running again. As it was a simple
+example application, this was thankfully straightforward and mainly required bumping versions of
+dependencies and a couple of cases of replacing unmaintained libraries. For a larger project this
+may well have been a major effort. The original code can be found on the [using-docker GitHub
+repository](https://github.com/using-docker/identidock) and the updated working version (prior to
+moving to Chainguard Images) can be found on the [v1
+branch](https://github.com/chainguard-dev/identidock-cg/tree/v1) of the [repository for this
+tutorial](https://github.com/chainguard-dev/identidock-cg/).
+
+In order to follow along with the tutorial, please clone the code and switch to the `v1` branch e.g:
+
+
+```bash
+git clone https://github.com/chainguard-dev/identidock-cg.git
+cd identidock-cg
+git switch v1
+```
+
+# Updating dnmonster
```

Comment:
What do you think about calling out the node rather than the project names? Also make sure this is an H2

```suggestion
## Updating the Node.js Microservice
```

Reply:
I like it. Done.